compiler/parser/parser.peg

```diff
@@ -63,67 +63,72 @@ Consts
 Const = __ v:ConstDef { RETURN(v) }
 
 ConstDef
-  = "const" _ id:IdentifierName __ "=" __ expr:Expr __ ";"? {
+  = "const" _ id:IdentifierName __ "=" __ expr:Expr {
       RETURN(MAP("name":id, "expr":expr))
     }
-  / "type" _ id:IdentifierName __ "=" __ typ:Type __ ";"? {
+  / "type" _ id:IdentifierName __ "=" __ typ:Type {
       RETURN(MAP(
         "name":id,
         "expr":MAP("kind":"TypeValue","value":MAP("kind":"TypeDef","name":id,"type":typ))))
     }
 
-Parallel
-  = __ "=>" __ s:Sequential __ ";" { RETURN(s) }
-
-FromTrunk
-  = __ source:FromSource  seq:FromTrunkSeq? __ ";" {
-      RETURN(MAP("kind": "Trunk", "source": source, "seq": seq))
-    }
-
-FromTrunkSeq
-  = __ "=>" __ seq:Sequential { RETURN(seq) }
-
-FromSource
-  = FileProc
-  / HTTPProc
-  / PassProc
-  / PoolBody
-
 Operation
-  = "split" __ "(" procArray:Parallel+ __ ")" {
+  = "split" __ "(" procArray:SplitLeg+ __ ")" {
       RETURN(MAP("kind": "Parallel", "procs": procArray))
     }
-  / "switch" _ expr:Expr _ "(" cases:SwitchLiteralClause+ __ ")" {
+  / "switch" _ expr:Expr _ "(" cases:SwitchLegs __ ")" {
       RETURN(MAP("kind": "Switch", "expr": expr, "cases": cases))
     }
-  / "switch" __ "(" cases:SwitchSearchBooleanClause+ __ ")" {
+  / "switch" __ "(" cases:SwitchLegs __ ")" {
       RETURN(MAP("kind": "Switch", "expr": NULL, "cases": cases))
     }
-  / "from" __ "(" trunks:FromTrunk+ __ ")" {
+  / "from" __ "(" trunks:FromLeg+ __ ")" {
       RETURN(MAP("kind": "From", "trunks": trunks))
     }
-  / Operator
+  / op:Operator
   / f:Function &EndOfOp { RETURN(f) }
   / a:OpAssignment &EndOfOp { RETURN(a) }
   / a:Aggregation &EndOfOp  { RETURN(a) }
   / expr:SearchBoolean !AggGuard {
       RETURN(MAP("kind": "Filter", "expr": expr))
     }
 
-SwitchLiteralClause
-  = __ expr:(DefaultToken / Literal / Identifier) __ "=>" __ proc:Sequential __ ";" {
+EndOfOp = __ (Pipe / SearchKeywordGuard / "=>" / [);] / EOF)
+Pipe = "|" !("{" / "[")
+
+SplitLeg
+  = __ "=>" __ s:Sequential { RETURN(s) }
+
+SwitchLegs
+  = __ first:SwitchLeg rest:(_ leg:SwitchLeg { RETURN(leg) })* {
+    RETURN(PREPEND(first, rest))
+  }
+
+SwitchLeg
+  = expr:Case __ "=>" __ proc:Sequential {
```

Comment:
You could use `SwitchLeg+` instead of `SwitchLegs` in the switch rules if you added a `__` prefix.
```suggestion
SwitchLeg
  = __ expr:Case __ "=>" __ proc:Sequential {
```